client/components/date-range/index.js

```diff
@@ -588,7 +588,7 @@ export class DateRange extends Component {
 	/**
 	 * Renders the Popover component
 	 *
-	 * @returns {ReactComponent} the Popover component
+	 * @returns {React.Component} the Popover component
```

Comment:
These are incorrect, they actually return `JSX.Element`.

```suggestion
	 * @returns {JSX.Element} the Popover component
```

Reply:
Indeed!